### PR TITLE
Hashicorp get specific secrets

### PR DESCRIPTION
**Pull request by gal-forer**
## Contributing to Cortex XSOAR Content
Make sure to register your contribution by filling the [contribution registration form](https://forms.gle/XDfxU4E61ZwEESSMA)

**The Pull Request will be reviewed only after the contribution registration form is filled.**

## Status
- [ ] In Progress
- [x] Ready
- [ ] In Hold - (Reason for hold)

## Related Issues
fixes: link to the issue

## Description
Added parameters to set the wanted secrets and the request type to the generate secrets API, and removed the ttl param from configure engine
## Screenshots
Paste here any images that will help the reviewer

## Minimum version of Cortex XSOAR
- [ ] 6.0.0
- [ ] 6.1.0
- [ ] 6.2.0
- [ ] 6.5.0

## Does it break backward compatibility?
   - [ ] Yes
       - Further details:
   - [ ] No

## Must have
- [ ] Tests
- [ ] Documentation

### CODE REVIEW

File: Packs/HashiCorp-Vault/Integrations/HashiCorpVault/HashiCorpVault.py [hunk 1]
```diff
@@ -607,7 +607,11 @@ def fetch_credentials():  # pragma: no cover
             credentials += get_ch_secrets(engine['path'], concat_username_to_cred_name)
 
         elif engine['type'] == 'AWS':
-            credentials += get_aws_secrets(engine['path'], engine['ttl'], concat_username_to_cred_name)
+            aws_roles_list = []
+            if engine.get('aws_roles_list'):
+                aws_roles_list = engine.get('aws_roles_list').split(',')
+            credentials += get_aws_secrets(engine['path'], concat_username_to_cred_name,
+                                           aws_roles_list, engine.get('aws_method', None))
```
Comment (DeanArbel) [inline Packs/HashiCorp-Vault/Integrations/HashiCorpVault/HashiCorpVault.py:614]: None is the default
```suggestion
                                           aws_roles_list, engine.get('aws_method'))
```

File: Packs/HashiCorp-Vault/Integrations/HashiCorpVault/HashiCorpVault_test.py [hunk 2]
```diff
@@ -31,10 +31,12 @@ def test_send_request(mocker):
...
 def test_get_aws_secrets(mocker):
     mocker.patch('HashiCorpVault.SERVER_URL', return_value='test')
-    get_aws_secrets('test', '999', False)
+    get_aws_secrets('test', False, None, None)
     assert mock.call_args.args[0] == 'test/roles'
-    get_aws_secrets('test', '999', False)
+    get_aws_secrets('test', False, None, None)
     assert mock.call_args.args[0] == 'test/roles/1'
+    # test aws_roles_list
+    assert get_aws_secrets('test', False, ['2'], None) == []
```
Comment (DeanArbel) [inline Packs/HashiCorp-Vault/Integrations/HashiCorpVault/HashiCorpVault_test.py:39]: Can you add a more meaningful test?

**Review by DeanArbel — APPROVED** [review]
Looks good, please see my notes.

**Comment by ShirleyDenkberg** [comment]
@DeanArbel Doc review completed.

**Pull request merged**